integration-libs/s4-service/order/components/cancel-service-order/cancel-service-order.component.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, inject } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { OrderEntry } from '@spartacus/cart/base/root';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  RoutingService,
+} from '@spartacus/core';
+import { OrderDetailsService } from '@spartacus/order/components';
+import {
+  CancelServiceOrderFacade,
+  CancelObj,
+} from '@spartacus/s4-service/root';
+import { mergeMap, Observable, throwError } from 'rxjs';
+
+@Component({
+  selector: 'cx-cancel-service-order',
+  templateUrl: './cancel-service-order.component.html',
+})
```

Comment:
```suggestion
@Component({
  selector: 'cx-cancel-service-order',
  templateUrl: './cancel-service-order.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
```
`major:` let's keep the convention and add a change detection strategy OnPush to prevent possible redundant re-rendering of this component and its child.